### ISSUE TITLE
Base_score for multiclass XGBoost in R

### ISSUE THREAD

**Issue by sakshays** [dmlc/xgboost#5824]
Im trying to figure out the working of XGBoost multiclass. Unlike binary classification the base_score which we provide is always overwritten by the algorithm to 1/(number of classes). I have noticed that the base_score is added to the final log_odds output. Any explanation on why this happens?

**Comment by trivialfis** [comment]
I'm not sure what do you mean.  base score is always a transformed score in XGBoost.

**Comment by sakshays** [comment]
What i meant is the base_score parameter in R lets you decide the starting probability for the model. In the case of binary classification as we change this parameter the gain and cover values changes accordingly, but in the case of multiclass problem the gain and cover values don't change irrespective of the starting probability. The starting probability is always 1/(number of classes). The base_score value we provide gets added to the final log_odds output

**Comment by hcho3** [comment]
The difference arises from the fact that XGBoost uses [one-vs-rest method](https://scikit-learn.org/stable/modules/multiclass.html#ovr-classification) to perform multiclass classification. So for each boosting round, we fit M trees, where M is the number of classes. The outputs from M trees are converted into a probability distribution by applying the softmax function.
